entity_api/src/organization.rs

```diff
@@ -81,19 +79,39 @@ pub async fn find_by(
         }
     }
 
-    Ok(query.distinct().all(db).await?)
+    // If no params provided, return all organizations
+    Ok(Entity::find().all(db).await?)
 }
 
 pub async fn find_by_user(db: &impl ConnectionTrait, user_id: Id) -> Result<Vec<Model>, Error> {
-    let organizations = by_user(Entity::find(), user_id).await.all(db).await?;
+    // Check if user is a super admin (has role = 'super_admin' with organization_id = NULL)
+    let is_super_admin = user_roles::Entity::find()
+        .filter(user_roles::Column::UserId.eq(user_id))
+        .filter(user_roles::Column::Role.eq(roles::Role::SuperAdmin))
+        .filter(user_roles::Column::OrganizationId.is_null())
+        .one(db)
+        .await?
+        .is_some();
+
+    warn!("User {} is_super_admin: {}", user_id, is_super_admin);
+
+    let organizations = if is_super_admin {
+        // Super admins have access to all organizations
+        let orgs = Entity::find().all(db).await?;
+        orgs
+    } else {
```

Comment:
Good, I like the else case being just a regular user here. Good secure design by default.